### ISSUE TITLE
Prop to Disable OK Button in Modal

### ISSUE THREAD

**Issue by sgoel1495** [kyndryl-design-system/shidoka-applications#89]
**Is your feature request related to a problem? Please describe.**

For a confirmation Modal, we need the user to type the name of the service before they can confirm its deletion.
The confirm button of the Modal should only become active once the name provided in the input field matches the name of the service.

**Describe the solution you'd like**

A prop with Boolean value to have a control over the disabled props of the ok button.

**Comment by brian-patrick-3** [comment]
This sounds like a use case for the `beforeClose` method that you can pass to the modal. It won't help you disable the button, but you can use it to perform the validations that you need to before it will let the modal close.

**Comment by sgoel1495** [comment]
Yes, right now I'm using `beforeClose` only. But would have been good to have the control to disable the primary button.
Keeping it enabled and then showing an error message that they need to fill in a field is not a user experience I would go for.
